src/node/transaction.cpp

```diff
@@ -21,15 +21,15 @@ TransactionError BroadcastTransaction(const CTransactionRef tx, std::string& err
 
     { // cs_main scope
     LOCK(cs_main);
+    // If the transaction is already confirmed in the chain, don't do anything
+    // and return early.
     CCoinsViewCache &view = *pcoinsTip;
-    bool fHaveChain = false;
-    for (size_t o = 0; !fHaveChain && o < tx->vout.size(); o++) {
+    for (size_t o = 0; o < tx->vout.size(); o++) {
         const Coin& existingCoin = view.AccessCoin(COutPoint(hashTx, o));
```

Comment:
Nit: add comment explaining `IsSpent()` doesn't mean the coin is spent; it means the output doesn't exist. So if the output does exist, then this transaction exists in the chain, i.e. it's already confirmed.

Reply:
Agree that a comment here would be helpful in a follow-up commit.

Reply:
Yes agree better documentation and refactoring of BroadcastTransaction can be pushed further in latter work.